rust/tw_evm/src/address.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright © 2017-2023 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+use std::fmt::{Display, Formatter};
+use std::ops::{RangeFrom, RangeInclusive};
+use std::str::FromStr;
+use tw_coin_entry::coin_entry::CoinAddress;
+use tw_coin_entry::error::{AddressError, AddressResult};
+use tw_encoding::hex;
+use tw_hash::{sha3::keccak256, H160};
+use tw_keypair::ecdsa::secp256k1;
+use tw_memory::Data;
+
+pub trait EvmAddress: FromStr<Err = AddressError> + Into<Address> {
+    /// Tries to parse an address from the string representation.
+    /// Returns `Ok(None)` if the given `s` string is empty.
+    #[inline]
+    fn from_str_optional(s: &str) -> AddressResult<Option<Self>> {
+        if s.is_empty() {
+            return Ok(None);
+        }
+
+        Self::from_str(s).map(Some)
+    }
```

Comment:
Is it possible to do something like `(!s.is_empty()).then(|| Self::from_str(s)).transpose()` ?

Reply:
I don't think so, @lamafab do you have any ideas?